src/Compilers/Core/Portable/SourceGeneration/GeneratorState.cs

```diff
@@ -25,6 +28,11 @@ namespace Microsoft.CodeAnalysis
                                                                          exception: null,
                                                                          elapsedTime: TimeSpan.Zero);
 
+        /// <summary>
+        /// A generator state for a generator that has not yet been initialized
+        /// </summary>
+        public static readonly GeneratorState Uninitialized = Empty with { Initialized = false };
```

Comment:
Honestly, not to nitpick even more, but this makes me think that the right answer is just to be resilient to `default(ImmutableArray)`, not to introduce a separate uninitialized state that is separate from `default`.